runtime/vm/exceptiondescribe.c

```diff
@@ -343,10 +343,25 @@ iterateStackTrace(J9VMThread * vmThread, j9object_t* exception, callback_func_t
 					pruneConstructors = FALSE;
 #endif
 					romClass = findROMClassFromPC(vmThread, methodPC, &classLoader);
-					if(romClass) {
-						romMethod = findROMMethodInROMClass(vmThread, romClass, methodPC);
-						if (romMethod != NULL) {
-							methodPC -= (UDATA) J9_BYTECODE_START_FROM_ROM_METHOD(romMethod);
+					if (NULL != romClass) {
+						J9UTF8 const *utfClassName = J9ROMCLASS_CLASSNAME(romClass);
+
+						ramClass = peekClassHashTable(vmThread, classLoader, J9UTF8_DATA(utfClassName), J9UTF8_LENGTH(utfClassName));
```

Comment:
`peekClassHashTable` will fail for non-bootclasspath classes defined in the SCC as the classLoader from `findROMClassFromPC` will always be the bootloader.  There is a large chance that SCC classes will be from the apploader as well so we should attempt to probe that loader as well before we fail back to the segment search.

Something like:

```c
if (ramClass == NULL) {
  if (j9shr_Query_IsAddressInCache(vm, romClass, romClass->size)) {
    /* Probe the application loader to determine if it has the J9Class for the current class.
     * This secondary probe is required as all ROMClasses from the SCC appear to be owned
     * by the bootstrap classloader.
     */
      ramClass = peekClassHashTable(vmThread, vm->applicationClassLoader, J9UTF8_DATA(utfClassName), J9UTF8_LENGTH(utfClassName));
  }
}
```

We'll still need a fallback path as this won't catch all cases but should catch substantially more.

Reply:
Thanks @DanHeidinga. I just tried this out and it is working fine. We are finding all the classes. I'll amend the PR with all the changes. 

However one minor issue stands. `j9shr_Query_IsAddressInCache` is declared as a static function in a header file [1]. This will make every compilation unit which includes this header to have a separate copy of that function. Moreover, if one includes that header file (SCQueryFunctions.h) then you cannot compile the compilation unit without errors because you have to use all the static functions defined in the header:

```
make[1]: Entering directory `/team/fjeremic/defects/iterateStackTrace/sdk/jre/lib/s390x/default/vm'
gcc -O3 -mtune=z10 -march=z9-109 -mzarch -g -DLINUX -D_REENTRANT -D_FILE_OFFSET_BITS=64 -fPIC -DIPv6_FUNCTION_SUPPORT  -DS390 -D_LONG_LONG -DJ9VM_TIERED_CODE_CACHE -fno-strict-aliasing -DS39064 -Wimplicit -Wreturn-type -Werror -Wall -I. -I../include -I../oti -I../util -I../gc_include -I../omr/gc/include -I../shared_common/include -I../gc_glue_java -I../nls -I../omr/include_core    -DTR_HOST_S390 -c -o exceptiondescribe.o exceptiondescribe.c
cc1: warnings being treated as errors
../shared_common/include/SCQueryFunctions.h:40: error: 'j9shr_Query_IsCacheFull' defined but not used
../shared_common/include/SCQueryFunctions.h:97: error: 'j9shr_Query_IsAddressInReadWriteCache' defined but not used
../shared_common/include/SCQueryFunctions.h:112: error: 'j9shr_Query_PopulatePreinitConfigDefaults' defined but not used
```

I think we may need to move these function definitions to a separate compilation unit or make them inline. Which do you prefer?

[1] https://github.com/eclipse/openj9/blob/b25d2d63793c8d12bba32f97c3848e4cb16409b7/runtime/shared_common/include/SCQueryFunctions.h#L73-L84

Reply:
@pshipton @hangshao0 Any preference on how we modify the SCQueryFunctions.h to deal with cases that only want to call 1 of the apis?  Or is there a better interface we should be using to check if a J9ROMClass is in the cache?

Reply:
I am fine with making them inline.

Reply:
Fixed in ddf8519 and 4d7bb30

Reply:
Post-approval nix - declaring functions (inline or not) in a .h file simply isn't the right thing to do. If the compiler (say, XLC) decides to ignore the inline directive, or someone wants to do a real debug build (disabling inlining entirely), this will fail.

Reply:
I'll withdraw the comment above (though not my objection about doing this in general). It follows the same pattern as j9accessbarrierhelpers.h.